src/components/EventUserTable.js

```diff
@@ -1,50 +1,166 @@
   }
 
+  /* updates stats and the rows in the table
+     faculty, gender, dietary, and year stats are only computed on the initial render of the component
+     # of registered/checkedin etc. is computed every single time this function is called
+  */
   async getEventTableData(eventID) {
-    const params = new URLSearchParams({
+    let params = new URLSearchParams({
+      eventID: eventID
+    })
+    await fetchBackend(`/registrations/?${params}`, 'GET')
+      .then(response => {
+        let heardFrom = {};
+        response.data.forEach(user => {
+          if (user.heardFromData) {
+            heardFrom[user.heardFrom] = heardFrom.hasOwnProperty(user.heardFrom) ? heardFrom[user.heardFrom] + 1 : 1;
+          }
+        })
+        this.setState({ heardFrom })
+      })
+
+    params = new URLSearchParams({
       users: true
     });
 
-    fetchBackend(`/events/${eventID}${params}`, 'GET')
-      .then(response => {
-        this.setState({ rows: response });
+    await fetchBackend(`/events/${eventID}?${params}`, 'GET')
+      .then(async users => {
+        this.registrationNumbers(users)
+        this.notRegistrationNumbers(users);
+      });
+  }
+
+  /**
+   * 
+   * @param {array of users} users
+   * calculates the stats for registration and updates the data for charts 
+   * each data set is an array of data (arrays) sets b/c different charts accept different data
+   */
+  async registrationNumbers(users) {
+    let registrations = {
+    }
+    users.forEach(user => {
+      if (user.registrationStatus) {
+        registrations[user.registrationStatus] = registrations[user.registrationStatus] ? registrations[user.registrationStatus] + 1 : 1;
+      }
+    })
+
+    this.setState({
+      rows: users,
+      registrations
+    });
+  }
+
+  /**
+  * 
+  * @param {array of users} users
+  * calculates any stats that aren't registration stats
+  * each data set is an array of data (arrays) sets b/c different charts accept different data
+  */
+  async notRegistrationNumbers(users) {
+    let faculties = {}, years = {}, dietary = {}, genders = {}
+    users.forEach(user => {
+      if (user.faculty) {
+        faculties[user.faculty] = faculties[user.faculty] ? faculties[user.faculty] + 1 : 1;
+      }
+      if (user.year) {
+        const yearInt = parseInt(user.year);
+        if (yearInt) {
+          years[yearInt] = years[yearInt] ? years[yearInt] + 1 : 1;
+        }
+      }
+      if (user.diet) {
+        dietary[user.diet] = dietary[user.diet] ? dietary[user.diet] + 1 : 1;
+      }
+      if (user.gender) {
+        genders[user.gender] = genders[user.gender] ? genders[user.gender] + 1 : 1;
+      }
+    })
+
+    this.setState({
+      faculties,
+      years,
+      genders,
+      dietary
+    });
+  }
+
+  async updateEventTableData(eventID) {
+    const eventParams = new URLSearchParams({
+      id: eventID
+    });
+
+    fetchBackend('/events/getUsers?' + eventParams, 'GET')
```

Comment:
Does this endpoint still exist?

Reply:
ya i missed that, idk why componentDidMount get's called when a registration is submitted. we also have that if statement in the componentDidUpdate that never gets called so this function is never used right now